blocks/bricks/recurrent.py

```diff
@@ -349,8 +348,7 @@ def state_to_reset(self):
     def get_dim(self, name):
         if name == 'mask':
             return 0
-        if name in (GatedRecurrent.apply.sequences
-                    + GatedRecurrent.apply.states):
+        if name in (self.apply.sequences + self.apply.states):
```

Comment:
I am not sure it is a nice change: what is the `apply` method is overridden in a descendant class?